pkg/csi_driver/multishare_controller_test.go

```diff
@@ -580,3 +588,467 @@ func TestGenerateInstanceDescFromEcfsDesc(t *testing.T) {
+	type OpItem struct {
+		id     string
+		target string
+		verb   string
+		done   bool
+	}
+	tests := []struct {
+		name          string
+		ops           []OpItem
+		initInstance  []*file.MultishareInstance
+		initShares    []*file.Share
+		req           *csi.DeleteVolumeRequest
+		resp          *csi.DeleteVolumeResponse
+		errorExpected bool
+	}{
+		// {
+		// 	name: "share not found, instance ready, instance deleted, succes response",
+		// 	req: &csi.DeleteVolumeRequest{
+		// 		VolumeId: testVolId,
+		// 	},
+		// 	resp: &csi.DeleteVolumeResponse{},
+		// },
+		// {
+		// 	name: "share not found, instance not ready (instance op in progress), error response",
+		// 	req: &csi.DeleteVolumeRequest{
+		// 		VolumeId: testVolId,
+		// 	},
+		// 	initInstance: []*file.MultishareInstance{
+		// 		{
+		// 			Name:     testInstanceName1,
+		// 			Location: testRegion,
+		// 			Project:  testProject,
+		// 			Labels: map[string]string{
+		// 				util.ParamMultishareInstanceScLabelKey: testInstanceScPrefix,
+		// 			},
+		// 			CapacityBytes: 1 * util.Tb,
+		// 			Tier:          "Enterprise",
+		// 			Network: file.Network{
+		// 				Ip: testIP,
+		// 			},
+		// 		},
+		// 	},
+		// 	ops: []OpItem{
+		// 		{
+		// 			id:     "op1",
+		// 			target: fmt.Sprintf(instanceUriFmt, testProject, testRegion, testInstanceName1),
+		// 			verb:   "update",
+		// 		},
+		// 	},
+		// 	errorExpected: true,
+		// },
+		// {
+		// 	name: "share not found, instance not ready (share op in progress for the instance), error response",
+		// 	req: &csi.DeleteVolumeRequest{
+		// 		VolumeId: testVolId,
+		// 	},
+		// 	initInstance: []*file.MultishareInstance{
+		// 		{
+		// 			Name:     testInstanceName1,
+		// 			Location: testRegion,
+		// 			Project:  testProject,
+		// 			Labels: map[string]string{
+		// 				util.ParamMultishareInstanceScLabelKey: testInstanceScPrefix,
+		// 			},
+		// 			CapacityBytes: 1 * util.Tb,
+		// 			Tier:          "Enterprise",
+		// 			Network: file.Network{
+		// 				Ip: testIP,
+		// 			},
+		// 		},
+		// 	},
+		// 	ops: []OpItem{
+		// 		{
+		// 			id:     "op1",
+		// 			target: fmt.Sprintf(shareUriFmt, testProject, testRegion, testInstanceName1, testShareName),
+		// 			verb:   "update",
+		// 		},
+		// 	},
+		// 	errorExpected: true,
+		// },
+		// {
+		// 	name: "share not found, instance ready with 0 shares, instance deleted, success response",
+		// 	req: &csi.DeleteVolumeRequest{
+		// 		VolumeId: testVolId,
+		// 	},
+		// 	initInstance: []*file.MultishareInstance{
+		// 		{
+		// 			Name:     testInstanceName1,
+		// 			Location: testRegion,
+		// 			Project:  testProject,
+		// 			Labels: map[string]string{
+		// 				util.ParamMultishareInstanceScLabelKey: testInstanceScPrefix,
+		// 			},
+		// 			CapacityBytes: 1 * util.Tb,
+		// 			Tier:          "Enterprise",
+		// 			Network: file.Network{
+		// 				Ip: testIP,
+		// 			},
+		// 		},
+		// 	},
+		// 	resp: &csi.DeleteVolumeResponse{},
+		// },
```

Comment:
what is the reason for these cases to be commented out?

Reply:
oops. fixed it